lib/addons/nginx/index.ts

```diff
@@ -97,13 +109,20 @@ export class NginxAddOn implements ClusterAddOn {
             }
         };
 
-        clusterInfo.cluster.addHelmChart("nginx-addon", {
+        const nginxHelmChart = clusterInfo.cluster.addHelmChart("nginx-addon", {
             chart: "nginx-ingress",
             repository: "https://helm.nginx.com/stable",
             release: Constants.SSP_ADDON,
             namespace: props.namespace,
             version: props.version,
             values
         });
+
+        const asyncTasks = Promise.all(dependencies.values()).then((constructs) => {
+            constructs.forEach((construct) => {
+                nginxHelmChart.node.addDependency(construct);
+            });
+        });
+        asyncTasks.catch(err => { throw new Error(err) });
```

Comment:
Ideally, I would like to refactor to some common addon implementation where you can specify in constructor the ids of dependencies and it will manage it automatically, as opposed to spreading similar code across multiple add-ons. E.g base class `BasicClusterAddOn` has an array of dependency names and . I am sure this will be used not just by us. 

A better approach to explore is to use a decorator pattern like https://www.typescriptlang.org/docs/handbook/decorators.html. I have not used it myself with typescript yet (used a lot with java) but it can look something like:

```
 class MyAddOn implements ClusterAddOn {
     
     @dependable('AwsLoadBalancerController')
     deploy(clusterInfo:  ClusterInfo): Promise<> {
      .....
     }
     
 }
```

We can defer this as well as functionally it serves the puprose (e.g. create a task to address later on).

Reply:
I've thought about the decorator pattern here as well (used it extensively in Python) but I too am new to it within the context of typescript. I will create a refactor task for it so we can unblock other PRs for now.